magento/magento2
# 31253 default store breadcrumbs returned for products which in both root categories

### Description (*)
Issue was related to the indexer catalog_category_product

### Related Pull Requests


### Fixed Issues (if relevant)
1. Fixes https://github.com/magento/magento2/issues/31253

### Manual testing scenarios (*)
Magento 2.4.1
1.Two root categories exists in Admin-Categories
2.Two stores exists with different root categories for each of it
3.Each Stores should have atleast one store view.
4.Each Root category should have one sub-category.
5.Product should belong to both stores. i.e. Assign product to all root and sub categories.
6.Run query
{
  "store": "engchildren"
}
query {
  products(filter: { url_key: { eq: "gabrielle-micro-sleeve-top"} }){
   	items {
      categories {
        name
        id
        url_path
        breadcrumbs {
          category_id
          category_name
          category_level
        }
      }
    } 
  }
}


### Questions or comments


### Contribution checklist (*)
